jobs/client-regeneration/client_regeneration/main.py

```diff
@@ -0,0 +1,383 @@
+import click
+
+from google.cloud import bigquery
+from datetime import datetime, timedelta
+
+DEFAULT_START_DATE = "2022-04-01"
+DEFAULT_END_DATE = "2022-06-01"
+
+COLUMN_LIST = [
+    "adjust_campaign",
+    "adjust_ad_group",
+    "adjust_creative",
+    "adjust_network",
+    "country",
+    "device_model",
+    "device_manufacturer",
+]
+
+
+def init_replacement_table(client, seed):
+    # init the table that will contain the mappings of client_ids to be replaced to their selected replacements
+    q = f"""CREATE OR REPLACE TABLE mozdata.analysis.regen_sim_client_replacements_{str(seed)} (
+        client_id STRING, -- client_id to be replaced - the "fake new" client that first showed up on `regen_date`.
+        label STRING, -- what we used to do the match.
+        regen_date DATE, -- the date the "fake new" client id shows up - will be replaced on this day forward.
+        regened_last_date DATE, -- the last date we observed the fake new client/the last day it needs to be replaced.
+        replacement_id STRING, -- the client_id we sampled from the churn pool to replace `client_id` above.
+        last_reported_date DATE, -- this is the day the replacement_id churned in the original history.
+        first_seen_date DATE -- this is the first_seen_date of the replacement_id churned in its original history.
+    )
+    PARTITION BY regen_date
+    """
+
+    job = client.query(q)
+    job.result()
+
+
+def init_regen_pool(client, seed, start_date):
+    # init a new version of the regen pool, write table to BQ and name it with the random seed that will be used in
+    # the sampling
+    # TODO: Can we get rid of this DECLARE?
+    q = f"""DECLARE start_date DATE DEFAULT DATE("{start_date}"); CREATE OR REPLACE TABLE 
+    mozdata.analysis.regen_sim_regen_pool_{seed} PARTITION BY regen_date AS SELECT * FROM 
+    mozdata.analysis.regen_sim_regen_pool_v1 WHERE regen_date >= start_date;"""
```

Comment:
i made a v2 version of the source table that goes back further, let's use that

```suggestion
    mozdata.analysis.regen_sim_regen_pool_v2 WHERE regen_date >= start_date;"""
```